app/include/drivers/behavior.h

```diff
@@ -23,6 +23,64 @@
  * (Internal use only.)
  */
 
+#if IS_ENABLED(CONFIG_ZMK_BEHAVIOR_METADATA)
+
+enum behavior_parameter_standard_domain {
+    BEHAVIOR_PARAMETER_STANDARD_DOMAIN_NULL = 0,
+    BEHAVIOR_PARAMETER_STANDARD_DOMAIN_HID_USAGE = 1,
+    BEHAVIOR_PARAMETER_STANDARD_DOMAIN_LAYER_INDEX = 2,
+    BEHAVIOR_PARAMETER_STANDARD_DOMAIN_HSV = 3,
+};
+
+struct behavior_parameter_value_metadata {
+    char *friendly_name;
+    uint8_t position;
+
+    union {
+        uint32_t value;
+        struct {
+            uint16_t min;
+            uint16_t max;
+        } range;
+
+        enum behavior_parameter_standard_domain standard;
+    };
+
+    enum {
+        BEHAVIOR_PARAMETER_VALUE_METADATA_TYPE_VALUE = 0,
+        BEHAVIOR_PARAMETER_VALUE_METADATA_TYPE_RANGE = 1,
+        BEHAVIOR_PARAMETER_VALUE_METADATA_TYPE_STANDARD = 2,
+    } type;
```

Comment:
There seems to be a bit of overlap between what this and `enum behavior_parameter_standard_domain` conceptually represent. If we can get rid of the special case in `behavior_parameter_metadata` as suggested below, maybe "domain" and "type" could be merged together?

Reply:
Done.